### PR TITLE
Update DB structure to fix archive_schedule_log function

### PR DESCRIPTION
**Pull request by louis-pie**
## Context

- Add tests for archive_schedule_log
- Fix DB structure
- Update Makefile to accept optional python version
- Clean some comments on existing tests

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: tests/test_functional_archive_logs.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM schedules WHERE is_enabled=1")[0][0]
+
+    assert results == 4
+
+
+def test_create_dummy_schedule_logs():
```
Comment (Samira-El) [inline tests/test_functional_archive_logs.py:125]: Why is this a test? this is more of a setup step that prepare data fixtures for main relevant test `test_archive_schedules`

File: tests/test_functional_archive_logs.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        ;
+        """
+    )
+
+    results = query_test_db(f"SELECT count(*) FROM servers WHERE is_enabled=1")[0][0]
+
+    assert results == 1
+
+
+def test_create_dummy_schedules():
```
Comment (Samira-El) [inline tests/test_functional_archive_logs.py:105]: Why is this a test? this is more of a setup step that prepare data fixtures for main relevant test `test_archive_schedules`

File: tests/test_functional_archive_logs.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+
+def test_test_db_setup(db_setup):
+    """test_test_db_setup"""
+    query_result = query_test_db("SELECT 1")
+
+    assert query_result == [(1,)]
+
+
+def test_create_dummy_servers():
```
Comment (Samira-El) [inline tests/test_functional_archive_logs.py:88]: Why is this a test? this is more of a setup step that prepare data fixtures for main relevant test `test_archive_schedules`

File: tests/test_functional_archive_logs.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    archive_schedule_log.main(1, pytest.db_test)
+
+    result = query_test_db(
+        f"SELECT (SELECT count(*) FROM schedule_log) || 'and' || (SELECT count(*) from schedule_log_historical)"
+    )[0][0]
+
+    assert result == '2and13'
+
+
+def test_archive_schedule_keep_correct_schedules():
```
Comment (Samira-El) [inline tests/test_functional_archive_logs.py:167]: the assert in this test should imo be included in `test_archive_schedules` where the actual archiving has been done. This test is not testing a different scenario, only the result of the previous one.

**Review by Samira-El — CHANGES_REQUESTED** [review]

File: tests/test_functional_archive_logs.py (hunk 1 above)
Comment (louis-pie) [reply tests/test_functional_archive_logs.py:125]: True, but having some check that the test data was inserted correctly has value.

File: tests/test_functional_archive_logs.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    """test_archive_schedule_keep_correct_schedules"""
+
+    result = query_test_db(
+        f"SELECT schedule_id, start_time, end_time, returncode FROM schedule_log"
+    )
+    print(result)
```
[the remark this answers is not in the document]
Comment (louis-pie) [reply tests/test_functional_archive_logs.py]: Thanks!

File: tests/test_functional_archive_logs.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        open(f"{pytest.cicada_home}/setup/schema.sql", "r", encoding="utf-8").read()
+    )
+    test_cur.close()
+    test_conn.close()
+
+
+def query_test_db(query):
+    """Run and SQL query in a postgres database"""
+    rows = []
+    with psycopg2.connect(
```
[the remark this answers is not in the document]
Comment (louis-pie) [reply tests/test_functional_archive_logs.py]: Changed to the more predictable way

**Pull request merged**